pkg/test/environment.go

```diff
@@ -146,4 +148,15 @@ func (env *Environment) Reset() {
 	env.LaunchTemplateCache.Flush()
 	env.SubnetCache.Flush()
 	env.SecurityGroupCache.Flush()
+
+	mfs, err := crmetrics.Registry.Gather()
```

Comment:
Can you explain what this is here for?

Reply:
Got it. Then, this might be better suited to the `ExpectCleanedUp` function. I'm fine with this here for now, but we should consider resetting metrics generally from the `karpenter-core` test package